Managers/UInputMan.cpp

```diff
@@ -1,2858 +1,1393 @@
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_L_UP].GetStick()].axis[element[INPUT_L_UP].GetAxis()].d2 = 0;
+						}
+					}
+				}
+
+				aimValues.Reset();
+
+				// Right stick
+				if (element[INPUT_R_LEFT].JoyDirMapped()) { aimValues.m_X = AnalogAxisValue(joystick, element[INPUT_R_LEFT].GetStick(), element[INPUT_R_LEFT].GetAxis()); }
+				if (element[INPUT_R_UP].JoyDirMapped()) { aimValues.m_Y = AnalogAxisValue(joystick, element[INPUT_R_UP].GetStick(), element[INPUT_R_UP].GetAxis()); }
+
+				if (aimValues.GetMagnitude() < deadZone * 2) {
+					if (element[INPUT_R_LEFT].JoyDirMapped()) {
+						if (joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].flags & JOYFLAG_UNSIGNED) {
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].pos = 128;
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].d2 = 0;
+						} else {
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].pos = 0;
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_R_LEFT].GetStick()].axis[element[INPUT_R_LEFT].GetAxis()].d2 = 0;
+						}
+					}
+					if (element[INPUT_R_UP].JoyDirMapped()) {
+						if (joy[joystick].stick[element[INPUT_R_UP].GetStick()].flags & JOYFLAG_UNSIGNED) {
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].pos = 128;
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].d2 = 0;
+						} else {
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].pos = 0;
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].d1 = 0;
+							joy[joystick].stick[element[INPUT_R_UP].GetStick()].axis[element[INPUT_R_UP].GetAxis()].d2 = 0;
+						}
+					}
+				}
+			}
+
+			// Stick, axis, directions
+			for (int stick = 0; stick < joy[joystick].num_sticks; ++stick) {
+				for (int axis = 0; axis < joy[joystick].stick[stick].num_axis; ++axis) {
+					// Adjust joystick values to eliminate values in deadzone
+					if (player > -1 && deadZoneType == DeadZoneType::SQUARE && deadZone > 0.0F) {
+						// This one heavily relies on AnalogAxiesValue method of processing joystick data.
+						if (joy[joystick].stick[stick].flags & JOYFLAG_UNSIGNED) {
+							if (std::abs(joy[joystick].stick[stick].axis[axis].pos - 128) > 0 && std::abs(joy[joystick].stick[stick].axis[axis].pos - 128) / 128 < deadZone) {
```

Comment:
Might be nice to pull the rather lengthy `joy[joystick].stick[stick].axis[axis].pos` out into a local variable, since it's used in a bunch of places.
If you wanna be super smort about it, make your local variable a pointer (or reference) to `joy[joystick].stick[stick].axis[axis]` so then you can set things on it in the lines that set stuff too.